### PR TITLE
feat(Sorryfield): add startTimestamp and chart (by song) list page

### PR DESCRIPTION
**Pull request by raiiiny**
## Description 

- added startTimestamp
- added chart (by song) list page
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image

</details>

### CODE REVIEW

File: websites/S/Sorryfield/presence.ts [hunk 1]
```diff
@@ -55,16 +57,20 @@ presence.on("UpdateData", async () => {
...
-				presenceData.endTimestamp = nowTime;
+						Math.floor(Date.now() / 1000) +
+						(parseInt(duration[0].replace("(-", "")) * 60 +
+							parseInt(duration[1].replace(")", ""))) +
+						1,
+					[startTimestamp, endTimestamp] = presence.getTimestampsfromMedia(
+						document.querySelector("video.back")
+					);
+				presenceData.startTimestamp = startTimestamp;
+				presenceData.endTimestamp = endTimestamp;
```
Comment (darkvillager2) [inline websites/S/Sorryfield/presence.ts:68]:
```suggestion
				[presenceData.startTimestamp, presenceData.endTimestamp] =  presence.getTimestampsfromMedia(
						document.querySelector("video.back")
					);
```

File: websites/S/Sorryfield/presence.ts [hunk 2]
```diff
@@ -101,16 +102,20 @@ presence.on("UpdateData", async () => {
...
 				if (nowTime <= Math.floor(Date.now() / 1000)) {
 					delete presenceData.endTimestamp;
 					presenceData.smallImageKey = Assets.Stop;
 				}
-			} else delete presenceData.endTimestamp;
+			} else delete presenceData.startTimestamp;
```
Comment (darkvillager2) [inline websites/S/Sorryfield/presence.ts:118]: huh?

**Review by darkvillager2 — COMMENTED** [review]
Hello,
Thanks for the Pull Request. I have a question about the code and i have requested changes to the code.

If you have any questions feel free to ask them here or on discord (@Dark_Ville)

File: websites/S/Sorryfield/presence.ts (hunk 1 above)
Comment (raiiiny) [reply websites/S/Sorryfield/presence.ts:68]: This suggest make some errors.
`ts(2451)` : `Cannot redeclare block-scoped variable 'presenceData'.`
`ts(2493)` : `Tuple type '[number, number]' of length '2' has no element at index '2'.`

I used Github Copilot to resolve these errors, but it reverses the code as it is now.

Comment (darkvillager2) [reply websites/S/Sorryfield/presence.ts:68]: you’re not compiling using our tool.

Comment (raiiiny) [reply websites/S/Sorryfield/presence.ts:68]: image

I used PreMiD's compile tool...

Comment (raiiiny) [reply websites/S/Sorryfield/presence.ts:68]: I found problem in line 106. I will commit it

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**